platforms/core-execution/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonServer.java

```diff
@@ -217,14 +223,39 @@ protected Instantiator createInstantiator(InstantiatorFactory instantiatorFactor
             return instantiatorFactory.decorateLenient(workerProjectScopeServices);
         }
 
+        /**
+         * ClientExecHandleFactory is available in global scope, but we need to provide it here to use a build dir based file resolver.
+         */
```

Comment:
❓ I don't understand this comment. If you put it here, the service will be available in build scope, not global, no?

Also, this class is already a workaround for some service loading. Do we need to add the service here as opposed to some other location? It would be good to not expand on existing workarounds.

Reply:
See discussion in https://github.com/gradle/gradle/pull/31318#discussion_r1848391094